src/xcode/ENA/ENA/Source/Scenes/Events/Store/MockEventStore.swift

```diff
@@ -0,0 +1,91 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import OpenCombine
+
+class MockEventStore: EventStoring, EventProviding {
+
+	// MARK: - Protocol EventStoring
+
+	func createTraceLocation(_ traceLocation: TraceLocation) -> VoidResult {
+		traceLocationsPublisher.value.append(traceLocation)
+		return .success(())
+	}
+
+	func deleteTraceLocation(id: String) -> EventStoring.VoidResult {
+		traceLocationsPublisher.value.removeAll { $0.guid == id }
+		return .success(())
+	}
+
+	func deleteAllTraceLocations() -> EventStoring.VoidResult {
+		traceLocationsPublisher.value = ([])
+		return .success(())
+	}
+
+	func createCheckin(_ checkin: Checkin) -> EventStoring.IdResult {
+		checkinsPublisher.value.append(checkin)
+		return .success((checkinsPublisher.value.count - 1))
+	}
```

Comment:
Don't we need a CheckIn struct without an id and some kind of auto increment mechanism?

Reply:
I thinks that's true for all other structs except the `TraceLocation`, as there the id comes from the server and is passed in.

Reply:
The TraceLocation struct has only a guid as the primary key. And the guid is delivered by the server.
But the Checkin struct needs an id which is the auto incremented primary key from the sql table.

Reply:
Ok, now I got it. So basically. If you want to create a Checkin you can just pass 0 as ID. It will not be used in case of creation.

Reply:
Another solution would be, to have a struct only for creation, without an id.

Reply:
That's what I had in mind. And we already have the `struct New` in e.g. the `DiaryLocation`, but did we forget to use it there? Looks like it 😅

Reply:
It feels also like a little to much overhead. I would just go with the id and not use it in case of creation.